### PR TITLE
IS-804: Implement hashCode and equals for EmbeddedKafkaContextCustomizer

### PR DESCRIPTION
**Pull request by OArtyomov**

### CODE REVIEW

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafkaContextCustomizer.java [hunk 1]
```diff
@@ -107,5 +107,19 @@ public void customizeContext(ConfigurableApplicationContext context, MergedConte
 		((DefaultSingletonBeanRegistry) beanFactory).registerDisposableBean(EmbeddedKafkaBroker.BEAN_NAME, embeddedKafka);
 	}
 
+	@Override
+	public int hashCode() {
+		return embeddedKafka.hashCode();
```
[the remark this answers is not in the document]
Comment (OArtyomov) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafkaContextCustomizer.java]: Done

File: spring-kafka-test/src/test/java/org/springframework/kafka/test/context/EmbeddedKafkaContextCustomizerTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.core.annotation.AnnotationUtils;
+
+
+
+/**
+ * @author Oleg Artyomov
+ * @since 1.3
+ */
+
+public class EmbeddedKafkaContextCustomizerTest {
```
[the remark this answers is not in the document]
Comment (OArtyomov) [reply spring-kafka-test/src/test/java/org/springframework/kafka/test/context/EmbeddedKafkaContextCustomizerTest.java]: Done

[the remark this answers is not in the document]
Comment (garyrussell) [reply spring-kafka-test/src/test/java/org/springframework/kafka/test/context/EmbeddedKafkaContextCustomizerTest.java]: Yes; agreed; this test is enough. Thanks for the contribution @OArtyomov

**Comment by artembilan** [comment]
... and cherry-picked to `2.1.x`, `2.0.x` and `1.3.x`.

@OArtyomov , thank you again!

**Pull request merged**